rasterio/_base.pyx

```diff
@@ -819,17 +819,71 @@ cdef class DatasetBase(object):
         num_items = CSLCount(metadata)
         return dict(metadata[i].split('=', 1) for i in range(num_items))
 
+
+    def get_tag_item(self, ns, dm=None, bidx=0, ovr=None):
+        """Returns tag item value
+
+        Parameters
+        ----------
+        ns: str
+            The key for the metadata item to fetch.
+        dm: str
+            The domain to fetch for.
+        bidx: int
+            Band index, starting with 1.
+        ovr: int
+            Overview level
+
+        Returns
+        -------
+        str
+        """
+        cdef GDALMajorObjectH band = NULL
+        cdef GDALMajorObjectH obj = NULL
+        cdef char *value = NULL
+        cdef const char *name = NULL
+        cdef const char *domain = NULL
+
+        ns = ns.encode('utf-8')
+        name = ns
+
+        if dm:
+            dm = dm.encode('utf-8')
+            domain = dm
+
+        if not bidx > 0 and ovr:
+            raise Exception("Band index (bidx) option needed for overview level")
+
+        if bidx > 0:
+            band = self.band(bidx)
+        else:
+            band = self._hds
+
+        if ovr:
+            obj = GDALGetOverview(band, ovr)
+            if obj == NULL:
+              return None
```

Comment:
@vincentsarago I think we should raise an exception here instead of returning `None`. A `rasterio.errors.BandOverviewError` would be good.

This will help us in debugging failures like https://travis-ci.org/mapbox/rasterio/jobs/367160869#L2701.

Reply:
good point!